lang/example.go

```diff
@@ -77,3 +77,8 @@ func (ex *Example) Code() (string, error) {
 
 	return code.String(), nil
 }
+
+// Output provides the code's example output.
+func (ex *Example) Output() string {
```

Comment:
This can't differentiate between examples which have an empty output and those which have no output defined. It looks like [`doc.Example`](https://golang.org/pkg/go/doc/#Example) provides an `EmptyOutput` field to help disambiguate.

Normally, I'd advocate for this having a return value of `(string, bool)`, but that doesn't play nice with templating. Maybe have a `HasOutput()` function alongside this one?

```go
func (ex *Example) HasOutput() bool {
	return ex.doc.Output != "" || ex.doc.EmptyOutput
}
```

Reply:
Implemented in latest commit. I originally tried only using `EmptyOutput`, but it was showing up as `false` even on examples that had no output. I opted for a tweak on this suggestion by ANDing the empty string check and bool check.

Reply:
From my reading of the docs, that's intentional. The `EmptyOutput` field specifically indicates when the example defines an expected output of `""`. So on an example with no output defined, I'd expect it to be false. Or am I misunderstanding the behavior you observed?

Reply:
OK, now I'm realizing that I misunderstood the purpose of `EmptyOutput`. I thought it was a signal that "there is no output", whereas it's actually a signal that "there is an output which is empty". In that case, I agree with your boolean logic, where we should have an empty output block in the case that `EmptyOutput` is true.

This is updated in the latest commit. I'll include samples below.